### DISCUSSION TITLE
Featurehub working in iOS but not in Android.

### DISCUSSION THREAD

**Discussion by NamrataDwivedi** [featurehub-io/featurehub#619] [General]
**Describe the bug**
Featurehub was working in Android before but now has stopped. Same code has been written for iOS and that is working till date. Upgrading the version of Feaurehub also not helpful.

 Which area does this issue belong to?
* [ ] SDK

**Expected behavior**
It should work for Android as well.

**Versions**
 - "featurehub-javascript-client-sdk": "^1.0.3"
 - "react-native": "0.63.2"
 - macOS Catalina 10.15.7 
 - docker-compose version is 1.27.0

**Comment by IrinaSouth** [comment]
Hi there, could you please let us know the following:
- what exactly the issue is, is there any error that you are seeing? Any code examples that causing it would be appreciated
- FeatureHub app version 

Thanks

**Comment by rvowles** [comment]
The problem not occurring on iOS's React Native runtime and occurring on Android's React Native runtime implies that the issue is with React Native and the way it is interacting, especially if everything else has remained the same. So any errors or traffic captures you can provide us would be helpful. We know there are issues with SSE on Android with React Native, but you have indicated these issues have suddenly appeared, so thats unlikely to be it.

So if I could add - are you using the GET method or SSE?

**Comment by NamrataDwivedi** [comment]
Hi @rvowles and @IrinaSouth ,

Current installed version is 1.2.0.

Attaching my code for you reference 

`import {
  EdgeFeatureHubConfig,
  Readyness,
  FeatureHubPollingClient
} from "featurehub-javascript-client-sdk";
import { APINETWORK } from "../constants/constants";

const edgeUrl = "https://abc.xyz.xux/features/";

const apiKey = APINETWORK.featurehub;
const fhConfig = new EdgeFeatureHubConfig(edgeUrl, apiKey);

const FREQUENCY = 500000;
fhConfig.edgeServiceProvider((repo, config) => new FeatureHubPollingClient(repo, config, FREQUENCY));
fhConfig.init();

let initialized = false;
console.log("Waiting for feature 0...");

export const FETCH_FEATUREHUB_REQUEST = "FETCH_FEATUREHUB_REQUEST";
export const FETCH_FEATUREHUB_SUCCESS = "FETCH_FEATUREHUB_SUCCESS";
export const FETCH_FEATUREHUB_ERROR = "FETCH_FEATUREHUB_ERROR";

const FeatureHubInitialState = {
  errorMsg: "",
  error: false,
  featurehubdata: [],
  isFetched: false,
  isCOHEnabled: false
};

export const fetchFeatureHub = () => {
  return {
    type: FETCH_FEATUREHUB_REQUEST,
  };
};
export const fetchFeatureHubSuccess = (action) => {
  return {
    type: FETCH_FEATUREHUB_SUCCESS,
    featurehubdata: action.featurehubdata,
    isCOHEnabled: action.isCOHEnabled
  };
};
export const fetchFeatureHubError = (action) => {
  return {
    type: FETCH_FEATUREHUB_ERROR,
    errorMsg: action.errorMsg,
  };
};

export const featureHubReducer = (state = FeatureHubInitialState, action) => {
  switch (action.type) {
    case FETCH_FEATUREHUB_REQUEST:
      return {
        errorMsg: "",
        error: false,
        featurehubdata: {},
        isFetched: false,
      };

    case FETCH_FEATUREHUB_SUCCESS:
      return {
        errorMsg: "",
        error: false,
        featurehubdata: action.featurehubdata,
        isFetched: true,
        isCOHEnabled: action.isCOHEnabled
      };

    case FETCH_FEATUREHUB_ERROR:
      return {
        errorMsg: action.errorMsg,
        error: true,
        featurehubdata: [],
        isFetched: true,
      };

    default:
      return state;
  }
};

export const fetchFeaturehubAsyncCreator = () => {
  return (dispatch) => {
    fhConfig.addReadynessListener(async (ready) => {
      console.log("at start feature 1");
      if (!initialized) {
        console.log("at start feature 2");
        console.log(`feature of Readyness =`, ready);
        if ((ready === Readyness.Ready)) {
          console.log("Feature are available, 3 starting server...");
          initialized = true;
          const fhClient = await fhConfig.newContext().build();

          if (fhClient) {
            console.log(`feature fhClient feature ===`, fhClient);
          }
        }
      }
    });
  };
};
`

![Screenshot 2021-12-09 at 10 13 27 AM](https://user-images.githubusercontent.com/19549471/145335919-df0d7a6f-c5b3-4161-bb3a-71f997b2766a.png)
b55fc22c.png)

Getting the mentioned warning in console and no api of featurehub getting called.
![Screenshot 2021-12-09 at 10 10 56 AM](https://user-images.githubusercontent.com/19549471/145337284-6b10ca80-999e-4f59-a9cd-7afeebc8d013.png)

In IOS the logs appear in following way in console : 
![Screenshot 2021-12-09 at 10 59 45 AM](https://user-images.githubusercontent.com/19549471/145339978-c1fbb398-9bde-49c8-9f1b-0ad88925401c.png)


You can also refer to my previous issue https://github.com/featurehub-io/featurehub/issues/477. After this all was working.
Not sure what went wrong.

**Comment by rvowles** [comment]
I'm not sure we can help you here I'm sorry - the fact that the SDK is exactly the same code, and it works on iOS indicates there is a problem somewhere else in the stack that is preventing the outgoing call from going ahead. I can only recommend that you debug into FeatureHubPollingClient on your Android client and see if you can see it making the request (this will be the init call) and find out what is going on. Please keep us in the loop as anything you find will be helpful to others!

**Reply by NamrataDwivedi** [reply]
It is always showing Not Ready, but it's same network and same code. Switching the network also did not work. Did you try at your end ? @rvowles
